addon.py

```
@@ -603,7 +603,34 @@ def download_polyhaven_asset(self, asset_id, asset_type, resolution="1k", file_f
+                                            algorithm = file_info['algorithm']
+                                            expected_checksum = file_info['checksum']
+                                            
+                                            # Compute actual checksum
+                                            try:
+                                                hash_obj = hashlib.new(algorithm)
+                                                hash_obj.update(content)
+                                                actual_checksum = hash_obj.hexdigest()
+                                                
+                                                # Verify checksum matches
+                                                if actual_checksum != expected_checksum:
+                                                    raise ValueError(f"Integrity check failed for {map_type}: checksum mismatch (expected: {expected_checksum}, got: {actual_checksum})")
```

Ruff (0.14.13)

Abstract `raise` to an inner function

(TRY301)